src/pybind/mgr/cephadm/module.py

```diff
@@ -3827,9 +3829,46 @@ def upgrade_resume(self) -> str:
     def upgrade_stop(self) -> str:
         return self.upgrade.upgrade_stop()
 
+    @handle_orch_error
+    def replace_device(self,
+                       hostname: str,
+                       device: str,
+                       yes_i_really_mean_it: bool = False) -> Any:
+        osds_to_zap: List[str] = []
+        if hostname not in list(self.inventory.keys()):
+            raise OrchestratorError(f'{hostname} invalid host.')
+
+        self.ceph_volume_list.get_data(hostname=hostname)
+
+        if device not in self.ceph_volume_list.all_devices():
+            raise OrchestratorError(f'{device} not a valid device in {hostname}.')
```

Comment:
it might be worth including in this error message to that a valid device is a device that appears in `ceph orch device ls` and is being used by at least one OSD.

Reply:
note that `ceph orch device ls` != `ceph-volume lvm list` but I get your point, i'll modify that message